backend/src/graphql/resolvers/BalanceResolver.ts

```diff
@@ -5,14 +5,30 @@ import { Resolver, Query, Ctx, Authorized } from 'type-graphql'
 import CONFIG from '../../config'
 import { Balance } from '../models/Balance'
 import { apiGet } from '../../apis/loginAPI'
+import { User as tUser } from '../../typeorm/entity/User'
+import { Balance as tBalance } from '../../typeorm/entity/Balance'
+import calculateDecay from '../../util/decay'
+import { roundFloorFrom4 } from '../../util/round'
 
 @Resolver()
 export class BalanceResolver {
   @Authorized()
   @Query(() => Balance)
   async balance(@Ctx() context: any): Promise<Balance> {
-    const result = await apiGet(CONFIG.COMMUNITY_API_URL + 'getBalance/' + context.sessionId)
+    // get public key for current logged in user
+    const result = await apiGet(CONFIG.LOGIN_API_URL + 'login?session_id=' + context.sessionId)
     if (!result.success) throw new Error(result.data)
-    return new Balance(result.data)
+
+    // load user and balance
+    const userEntity = await tUser.findByPubkeyHex(result.data.user.public_hex)
```

Comment:
Instead of findByHex you could also use the email here. Where is the benefit?

Reply:
> Great - tho the balance call is way less code then expected. This is a great step!

Yes but this has to do with ignoring a big peace of code entirely. 
I don't use the auto-error-correction code from community-server. It seems that the decay calculation is now stable so we maybe don't need at any longer. But it could lead to errors in future.

Reply:
At the beginning sometimes the key-pair from users changed and so multiple users with the same email exist. 
The transactions themself ultimately are based on the public key not on the email. 
So it is more robust that way. 
And it prevent data loss when a key-pair from a user change. 
But it isn't ideal we maybe can optimize it further.